railties/lib/rails/generators/primary_file_helpers.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Rails
+  module Generators
+    module PrimaryFileHelpers # :nodoc:
+      def self.included(base)
+        base.class_option :editor, type: :string, aliases: "-e", lazy_default: ENV["EDITOR"],
+                                   required: false, banner: "editor",
+                                   desc: "Open generated primary file in the specified editor (Default: $EDITOR)"
```

Comment:
What type of generators can't support the --editor option?

Reply:
For example, it can not support scaffold generator which can not decide primary file.